src/components/Eliza/CoreEliza.tsx

```diff
@@ -0,0 +1,290 @@
+import React, { useCallback, useMemo } from 'react';
+
+import ChooseCharacterFile from './components/ChooseCharacterFile';
+import EditCharacterFile from './components/EditCharacterFile';
+import ConfigureEnvFile from './components/EditEnvFile';
+import Step from './components/Step';
+import ChooseEnvFile from './components/ChooseEnvFile';
+import ConfirmAgentDetails from './components/ConfirmAgentDetails.tsx';
+import DeploymentStatus from './components/DeploymentStatus.tsx';
+import DeploymentFailed from './components/DeploymentFailed.tsx';
+
+import {
+  useDeployAIAgent,
+  type UseDeployAIAgentProps,
+} from './hooks/useDeployAIAgent.ts';
+
+interface StepConfig {
+  id: string;
+  condition: boolean;
+  content: React.ReactElement;
+}
+
+interface ElizaCoreProps {
+  isLoggedIn: UseDeployAIAgentProps['isLoggedIn'];
+  login: UseDeployAIAgentProps['login'];
+  triggerAgentDeployment: UseDeployAIAgentProps['triggerAgentDeployment'];
+  getAgentDeploymentStatus: UseDeployAIAgentProps['getAgentDeploymentStatus'];
+  ensureUserSubscription: UseDeployAIAgentProps['ensureUserSubscription'];
+}
+
+export const CoreEliza: React.FC<ElizaCoreProps> = ({
+  isLoggedIn,
+  login,
+  triggerAgentDeployment,
+  ensureUserSubscription,
+  getAgentDeploymentStatus,
+}) => {
+  const elizaIntegrations: UseDeployAIAgentProps = {
+    login,
+    isLoggedIn,
+    triggerAgentDeployment,
+    getAgentDeploymentStatus,
+    ensureUserSubscription,
+  };
+
+  const {
+    activeStep,
+    goToPreviousStep,
+    goToNextStep,
+    characterFile,
+    setCharacterFile,
+    envFile,
+    setEnvFile,
+    isDeploymentSuccessful,
+    isDeploymentFailed,
+    deploymentStatus,
+    fleekMachineUrl,
+  } = useDeployAIAgent({ ...elizaIntegrations });
+
+  const GoBackButton = (
+    <button
+      className="text-[14px] leading-[20px] text-[#F5E147]"
```

Comment:
Let's try to prevent hard types.

```suggestion
      className="text-[14px] leading-[20px] text-yellow-dark-11"
```

Reply:
I am not sure if those tailwind tokens will match the same colors when this is moved to another repo and imported into different projects